samples/loadimpactair_script_sample.js

```diff
@@ -0,0 +1,397 @@
+﻿// This script is a user journey for http://loadimpactair.guldskeden.se
+
+// LOAD IMPACT AB
+// 
+// Copyright 2017 Load Impact AB
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.See the License for the specific language governing permissions and limitations under the License.
+//
+
+import { group, sleep } from "k6";
+import http from "k6/http";
+import { Trend } from "k6/metrics";
+
+/**
+ * Returns random number between (inclusive) min and max
+ */
+function rnd(min, max) {
+  min = Math.ceil(min);
+  max = Math.floor(max);
+
+  return Math.floor(Math.random() * (max - min + 1)) + min;
+}
+
+/**
+ * Returns URL encoded string
+ */
+function urlencode (str) {

```

Comment:
Remember, the code was originally generated from a user journey recording. The POSTs were not hand written.

It is highly unlikely anyone will _ever_ script lengthy user journeys manually. 

Bodies look like this:

``` "ctl00%24MainContent%24ctl00=ctl00%24MainContent%24ctl06%7Cctl00%24MainContent%24ctl08&ctl00%24MainContent%24ddlDeparture=1&ctl00%24MainContent%24ddlDestination=3&__EVENTTARGET=&__EVENTARGUMENT=&__LASTFOCUS=&__VIEWSTATE=0siwmMXNLywURZs%2Bg%2Fji8J9chvfrLe%2BvfHK1s2SCxclI7uuENpGUtHiNlFYsCBeQKDVFzoeIVRZWlNaabZfNIvoAUD2xMbIoBASGMgVFTSFTqYzI0X3esqzSzAMGMMkdOGbE36kQX983v3Yk4CZKt4nJ6KvN64Pw%2B5q7awSFf0AzKr%2BabVitsXIAtwSWMX3MGvIl7QUTn4yZtsRuqsga1m9evmWHADBoJAucX0VosWxhI1q8HWY3hTzoqEx0cOEmc7j3yTrM0m0rlUhNV55CTm%2FLrzgTTnOPvMAVEemfWY0O0iuUZZOnZTMFWbDRCIXDwoDub%2BEkHZ22FIWZzlCUwg6cZ1sw%2Fg34roe9mVDaDTESTMvg6ZLdMObqsfpf5%2FVq2AN0W%2BL6amT9dKjV7FOHaBxzl37L%2Ft6sCKxaMUXXg9T6Nkwj7sLiyXpDRLPLmmKmOzhOU5AElLyYg2fg77AvI7jadEcib7bMXl1VfUu27EdQTt2Bfjp21Mv8EDG99l23nVelpMMxiToMk%2F2s2%2BWTyFuhv20bYnmsFqm1OABPjIcCb79%2FUo743IVqk3zMAnppJyS5SemW6aAM7%2BE1Zl%2BkV7c8hW%2BOprxaBZR%2BTzY1%2B6hSxwBOzhKmlEHtjsUqwa5yQniNx62gis4yXhaBb8ssp5MQNO4xAvV%2Bwgo6%2FnV0srbjvDcNfy0%2BccTf0jhLp53W&__VIEWSTATEGENERATOR=A4196A29&__EVENTVALIDATION=dMdSbDUdIWCVR3BUxTHe%2B5SkF9PeM7rRcJS6ml3CndPlHV4j717H16jbBvddhTgDDM6ewiaa46ry%2BvCZeFRYZztvTWZEAiZyNLVeGFG8IK9h1srytBPjZ%2F16QcBBuCEEe8eEsq5%2BGfCSgMmtp7g36wCxPmhdCBGh4Sa9i5ujoJ8CCY6Gj5RcgiI0BBQo6JlmVYPvWexWjkMCkZChbXXj7qTuAablBqay%2BXVKtj6IG9n9Nx%2BiMg6Ki1zCaf2g%2FSDAW9NfX6iWGByLsHnk4UyCKh4H8GQQTXOvzHZ9ZFU2BMCjAkqdygvj48g24dU0uSJ6kgQWPx0UEIHAQaO%2F6JUER6Zoze9XiRB5gc3whtMGqZ56x9IwDR%2BoahA6dv9Tqh387ovz4YkgLnJoyGtqItnZSHFE9GrOuZ5UjWshLo7iEtCZINFjfboS0WROCr2w6q5Mhhf107nXpNxUbF9Ox%2FxiUOkUvNzWwLpYf%2FKGE3EGz6rgJ%2FezlytTHvWbdjWVw5rB1OftpGSIr4X6sx06lLQ2qF1J2rR9FUzQbA%2B4dnIuKrsUixft546mOCHBA%2B7cPGpXslA1Z83q0KKXmGHQiF67LWKbJXcJWcSb4tb2a%2FK5vfwwLHoiEdgE3BN5MQtiZz1zJduPccRTH597DiBaLRPvAdA4llxw8MO%2FFEZduxBZttLw7j%2FJA7VOmNAmy9fuFFvs49mJoGKcItfHkIRt0otId3ZPrbyLn1mry6Vl1REceJy0STr4WCRU0Ilfbm4Hn7%2B7O%2BdZBZN3mdwtwkYe5J%2FiKgLl%2FzWA7ZqnJVQE%2Bsuo%2FePHGwWX6FmNBB%2FerDsqyjdMw94b98ecXTUg2lkxqOC2n1g0IO011OxFEws%2F1Fe7%2FCh%2BgDuD7fgpwNDJ%2F9LZ1OIiQBZ8%2Fs3uHqZpUX8CqJXI310TjvzIKFUjrqbpOK3dU2ltaV3f1IC4c8G7DfbyVM0TyszfOi%2BL%2F2zZBHDQXGvB7i2cXuViHYJvDIdHXy80l0pp2GeW0NGpI68%2BoPg6z09gEfdZMCfx59asDGBt0opduRp1wfvAe2B9Zl6I7Zqay%2B%2Bqz2hsqUG%2Bx%2BcnzUjaVIRl4shNp0j68Oy5Xx6jfl885sw95s4IFQC0sTZCKM5a0xaAoHCWX4JKKKCMZIIq9EmbgScnHmKEo1KYQFaxzEPeUbYzkyusCN8lpyAPfisdfMYjwETzuux8w29WP%2FWEes0aXvGv7QB61woGeSMGzeIWriUvgGjkSdx%2Be0VBnLxQ%2BTlG4G16vyZ%2BGuuKFR73ePCh0UHyxbymJDq6T%2BBRfLwCDbJF09X0onGcN6d4GKT6is8zaHr3zlGBynNzqIs1SwYrzPL9wK1AFq0aQx6B5x4vZyc8QpV0HThPV7jIi14oerQZ5HhBJoAq86Aphbi4u01s0KhjirOlPvVFExlqFTcpmth4suFY70lzGGvSH9NABSkezA5KqrvWM9HXYlF7h9YMSiD46qvt7tRYZQtLFHX04Gsvp7ghraBz0op8KIUmzZ0iHQoiiFjrEn6IrMGYzSykE4%2FPYWFW7NVFDvWXRoloRQqypCQxoSHeHgI5tsxQX%2BJzWkNUqMeOVmpQ72E4CttUnrSO%2BxseIzWiEBLX5DXH7syv1uroQwEbsNpqEFJaDfcEfGItkVrz74K8BQhTqPRwipZ%2B2sjKyx48AbC%2FYlfNtHDmUWQvcBTD4aV%2BroPF2adEfybRV6Nrr7cSY5K7o%2F1AIEigZ4LyZXWQrZTMrbcqiJHRuPFdoD4DVMMhrXRJ4hSyXXIzopSGLaCadVyDiwS7AUDc6K4OSB8k%2BRw5pA9d1auN2tQYLKMZ1hntTiavdPYB1yI1aFnlhFzAXWVMsWSkBWmbkiP5BncTpdVWh5O8ZCBs0Jv1vl%2Bp7NMhX%2Ba4L7z%2FNpqAoiqZsjexoKJXERYDsSsDf4VhQlEvMDYI%2BT7ALD%2BuVdndFP5yNtmPfyoSD6zyqziFHy1RQzO76Bak571mdhqSy5zckz%2Be9RHIZPVnWRWBnXQ5OKbt7oOaW6HI4JSGqzEO1MvN0kRN17ttY6zwishKcF3M5i%2Brc0tvMJ9eRHkArr5s%2Foy7Ell7JKTwmnK9uBeweXc8nckQUIsouODxSKhtFJoGlbRoz4HhP5F7wNl4IPTq%2FJv1LQJfDEUfCBFHzGFabInCiaCR%2BcvROqFCv5VikqN73MtkFSre%2FvAxzc2r0KMnBH7U4lnmip5W6rycq3huzYkwp4qBtycPUK%2FNLv5GqRUDTFF%2FOYKU3lgU%2BWbTyY%2BFdg26o9SHxSsOMzQdzw65Xs0gA7rr84IpMVwL&__ASYNCPOST=true&ctl00%24MainContent%24ctl08=Search%20flights"```

Manually writing code to split, url-decode and use bodies as objects is a lot of extra work for no obvious extra value.

For the purpose of providing useful examples - it's not needed. For the purpose of providing _perfect_ examples it could be.

But of course (off on a slight tangent) - our exporter [Fiddler to Load Impact](https://github.com/loadimpact/FiddlerToLoadImpact) could do it automagically for users. Inlining as objects rather than strings would be simple enough.

What about multipart then like this?

```js
{"method" : "POST", "url" : "https://dm2305.storage.live.com/mydata/windows/settings?SyncToken=LM%3d63619208892620%3bID%3dE03F0AB241D3E649!111%3bLR%3d63619209049033%3bEP%3d13%3bSI%3d84%3bTD%3dTrue%3bSO%3d0%3bPI%3d49&View=WindowsSettingsV2", 
"body" : `


--{84FDFF35-F926-4B9D-9599-558815628CBE}
Content-Type: application/web3s+xml
Content-ID: Settings

<?xml version=\"1.0\" encoding=\"UTF-8\"?><Items><Folder><RelationshipName>windows-slideshow</RelationshipName><RoleDefinitionName>WindowsSettingsFolder</RoleDefinitionName><WriteMode>CreateOrUpdate</WriteMode><Items><Document><RelationshipName>SlideShow</RelationshipName><ETag>E03F0AB241D3E649!147.382</ETag><WriteMode>Update</WriteMode><DateModifiedOnClient>2017-01-05T09:23:42.929Z</DateModifiedOnClient><windowssettings.xschema.drx.live.com><CreationId>0</CreationId></windowssettings.xschema.drx.live.com></Document></Items></Folder></Items>

--{84FDFF35-F926-4B9D-9599-558815628CBE}
Content-Type: application/octet-stream
Content-ID: <windows-slideshow/SlideShow:Binary>
Content-Transfer-Encoding: binary

` + bin1 + `

--{84FDFF35-F926-4B9D-9599-558815628CBE}--`
, "params" : { headers: { "Content-Type" : "multipart/related;boundary=\"{84FDFF35-F926-4B9D-9599-558815628CBE}\""} }},
```